### PR TITLE
Cleanup `#pragma` and `unsigned char` warnings on some platforms

### PR DESCRIPTION
**Pull request by kcgen**
Some platforms used an unsigned type for `char`, leading to warnings like:

``` text
  ../src/dos/drive_cache.cpp: In function ‘Bits wine_hash_short_file_name(char*, char*)’:
  ../src/dos/drive_cache.cpp:493:16: warning: comparison is always false due to limited range of data type [-Wtype-limits]
    493 |   *dst++ = (*p < 0 || strchr( invalid_chars, *p ) != NULL) ? '_' : toupper(*p);
        |             ~~~^~~
  ../src/dos/drive_cache.cpp:508:19: warning: comparison is always false due to limited range of data type [-Wtype-limits]
    508 |    *dst++ = (*ext < 0 || strchr( invalid_chars, *ext ) != NULL) ? '_' : toupper(*ext);
        |              ~~~~~^~~
```

### CODE REVIEW

File: include/support.h [hunk 1]
```diff
@@ -53,6 +53,15 @@
...
 
+constexpr bool char_is_negative([[maybe_unused]] char c)
+{
+#if (CHAR_MIN < 0) // char is signed
+	return c < 0;
+#else // char is unsigned
+	return false;
+#endif
+}
+
```
Comment (kcgen) [inline include/support.h:64]: I originally used [std::is_unsigned](https://en.cppreference.com/w/cpp/types/is_unsigned) from `<type_traits>`:

``` c++
  return std::is_unsigned<char>::value ? false : char < 0;
```

But if `char` really is unsigned, then the `char < 0` will never be evaluated and the compiler will warn about that second condition never being evaluated.

File: src/dos/drive_cache.cpp [hunk 2]
```diff
@@ -468,44 +468,70 @@ bool DOS_Drive_Cache::RemoveTrailingDot(char* shortname) {
...
+		                                  ',',
+		                                  ';',
+		                                  '[',
+		                                  ']',
+		                                  ' ',
+		                                  '\345',
+		                                  '~',
+		                                  '.',
+		                                  '\0'};
+		const auto is_invalid = char_is_negative(c) ||
```
Comment (johnnovak) [inline src/dos/drive_cache.cpp:492]: Looks funky, but OK. At least it's formatted automatically :)

**Review by johnnovak — COMMENTED** [review]
Nice cleanup job 👍🏻

**Comment by kcgen** [comment]
Thanks for the review, @johnnovak .

**Pull request merged**